### DISCUSSION TITLE
What is the recommended way to run k8s jobs from dagster (with docker deployment)

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18228] [Q&A] [answered]
Hi! What is the best way to run k8s job from dagster (with docker deployment)? Needed to run spark applications in spark-on-k8s-operator, i'm currently using command line and `kubectl apply manifest.yaml`.
Found `execute_k8s_job`  at <https://docs.dagster.io/_apidocs/libraries/dagster-k8s>, but not shure, that its what i need.

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1700640600044239?thread_ts=1700640600.044239&cid=C01U954MEER).*

**Comment by gibsondan** [comment, marked as answer]
You may find Dagster Pipes a good fit for this use case: https://docs.dagster.io/guides/dagster-pipes/kubernetes#integrating-kubernetes-with-dagster-pipes
